File: tests/java/tests/java/org/elastos/hive/config/ClientConfig.java
package org.elastos.hive.config;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ClientConfig extends Config {
	@JsonProperty("resolverUrl")
	private String resolverUrl;
	@JsonProperty("application")
	private ApplicationConfig applicationConfig;
	@JsonProperty("user")
	private UserConfig userConfig;
	@JsonProperty("node")
	private NodeConfig nodeConfig;
	@JsonProperty("cross")
	private CrossConfig crossConfig;

	public String resolverUrl() {
		return this.resolverUrl;
	}

	public ApplicationConfig applicationConfig() {
		return this.applicationConfig;
	}

	public UserConfig userConfig() {
		return this.userConfig;
	}

	public NodeConfig nodeConfig() {
		return this.nodeConfig;
	}

	public CrossConfig crossConfig() {
		return this.crossConfig;
	}

	public static ClientConfig deserialize(String content) {
		return deserialize(content, ClientConfig.class);
	}
}

File: src/config/clientconfig.ts
import { ApplicationConfig } from "./applicationconfig"
import { UserConfig } from "./userconfig"
import { CrossConfig } from "./crossconfig"
import { NodeConfig } from "./nodeconfig"
import { JSONObject } from "@elastosfoundation/did-js-sdk/typings";

export class ClientConfig {
	private resolverUrl: string;
	private application: ApplicationConfig;
	private user: UserConfig ;
	private node: NodeConfig ;
	private cross: CrossConfig ;

	public getResolverUrl(): string {
		return this.resolverUrl;
	}

	public getApplicationConfig(): ApplicationConfig {
		return this.application;
	}

	public getUserConfig(): UserConfig {
		return this.user;
	}

	public getNodeConfig(): NodeConfig {
		return this.node;
	}

	public getCrossConfig(): CrossConfig {
		return this.cross;
	}

	public static deserialize(json: any): ClientConfig {
		let jsonObj = (typeof json === "string" ? JSON.parse(json as string) : json);
		let newConfig = new ClientConfig();
		newConfig.resolverUrl = jsonObj['resolverUrl'];
		newConfig.application = ApplicationConfig.deserialize(jsonObj['application']);
		newConfig.user = UserConfig.deserialize(jsonObj['user']);
		newConfig.node = NodeConfig.deserialize(jsonObj['node']);
		newConfig.cross = CrossConfig.deserialize(jsonObj['cross']);

		return newConfig;
	}
}
